homeassistant/components/arve/coordinator.py

```diff
@@ -0,0 +1,62 @@
+"""Coordinator for the Arve integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+
+from asyncarve import Arve, ArveConnectionError, ArveDevices, ArveError, ArveSensProData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+
+class ArveCoordinator(DataUpdateCoordinator[ArveSensProData]):
+    """Arve coordinator."""
+
+    config_entry: ConfigEntry
+    devices: ArveDevices
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize Arve coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=60),
+        )
+
+        self._client_session = async_get_clientsession(hass)
+
+        self.arve = Arve(
+            self.config_entry.data[CONF_ACCESS_TOKEN],
+            self.config_entry.data[CONF_CLIENT_SECRET],
+            session=self._client_session,
+        )
+
+        self.first_start = True
+
+    async def _async_update_data(self) -> dict[str, ArveSensProData]:
+        """Fetch data from API endpoint."""
+        try:
+            if self.first_start:
+                self.devices = await self.arve.get_devices()
+                self.first_start = False
+
+            response_data = {
+                sn: {
+                    "sensors": await self.arve.device_sensor_data(sn),
+                    "info": await self.arve.get_sensor_info(sn),
+                }
+                for sn in self.devices.sn
+            }
```

Comment:
I think you can keep polling the devices every so often, checkout tedee for an example. A future improvement (for a followup PR) is automatically registring and removing devices if we can't find them, tedee has a good setup for that)